src/matrixrtc/CallMembership.ts

```diff
@@ -19,19 +19,84 @@ import { deepCompare } from "../utils";
 import { Focus } from "./focus";
 
 type CallScope = "m.room" | "m.user";
-
 // Represents an entry in the memberships section of an m.call.member event as it is on the wire
-export interface CallMembershipData {
-    application?: string;
+
+// There are two different data interfaces. One for the Legacy types and one complient with MSC4143
+
+// Current MSC4143 session membership data
+export interface SessionMembershipData {
+    application: string;
+    call_id: string;
+    device_id: string;
+
+    foci_active: Focus;
+    foci_preferred: Focus[];
+    created_ts?: number;
+
+    // Application specific data
+    scope?: CallScope;
+}
+const isSessionsMembershipData = (data: CallMembershipData): data is SessionMembershipData =>
+    "foci_active" in data &&
+    "foci_preferred" in data &&
+    "membership_id" in data &&
+    !Array.isArray(data.foci_active) &&
```

Comment:
Wouldn't "is string" be closer?

Reply:
Also, you're checking this isn't an array but the docs (and test examples) say it should be, so I'm confused.

Reply:
There are two versions of the CallMembershipData.
SessionMembershipData is the new one enforcing foci active to be a object.
`{"type":"livekit", "livekit_service_url":"https://sfu.abc"}`
Here the foci_preferred takes to role of listing all avialable foci.


And CallMembershipDataLegacy where foci_active is the array of availble foci.